### PR TITLE
Add new qvm-block implementation

### PR DESCRIPTION
**Pull request by kalkin**
Notable/Useful Additions:

- VolumeAction for parsing POOL_NAME:VOLUME_ID combinations
- Add Volume.verify()
- Add subparser handling to parse_qubes_app

### CODE REVIEW

File: qubes/tools/qvm_block.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+        except AttributeError:
+            # Skipping domain without volumes
+            continue
+
+    for domain in domains:
+        if domain.qid == 0:
+            continue
+        lvirt = domain.libvirt_domain
+        xml = etree.fromstring(lvirt.XMLDesc())
+        disks = xml.xpath("//domain/devices/disk")
```
Comment (woju) [inline qubes/tools/qvm_block.py]: This probably should be done in VM (with some dummy method overload in AdminVM) and only aggregated here. I'd imagine Qubes Manager would also need this parsing.

As a rule of thumb, qvm-tools shoudn't parse XML, especially from libvirt.

File: qubes/tools/qvm_block.py [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+        else:
+            output += [(str(volume), volume.volume_type)]
+
+    return output
+
+
+class VolumeData(object):
+    ''' Wrapper object around :py:class:`qubes.storage.Volume`, mainly to track
+        the domains a volume is attached to.
+    '''
```
Comment (woju) [inline qubes/tools/qvm_block.py]: Why can't this me merged with `qubes.storage.Volume` or at least put somewhere in `qubes.storage` (for example as mixin)? As below, Qubes Manager.

Comment (woju) [inline qubes/tools/qvm_block.py]: To clarify, this is mostly about `.domains` attribute.

File: qubes/tools/qvm_block.py (hunk 1 above)
Comment (kalkin) [inline qubes/tools/qvm_block.py]: FULL ACK, will change this

File: qubes/tools/qvm_block.py (hunk 2 above)
Comment (kalkin) [inline qubes/tools/qvm_block.py]: I would ask if we could merge this as it is for **now**. 

> Why can't this me merged with qubes.storage.Volume or at least put somewhere in qubes.storage (for example as mixin)?

1. I would prefer not to save information about the domain in a `Volume`.

2. Once i get rid of the the file backend and make LVM first kind citizen, the way how we manage Volumes will slightly change. Currently we have on `vid` to identify the snapshot of a specific `'origin'`. In future i want to make `vid` to be always unique. The physical volume will know from which volume it was cloned/snapshoted (This is not possible with files, but with LVM, Btrfs …). So i will be forced to touch this code again.

3. I just want to have for now a clean simple git tree to work further.

**Pull request merged**